### PR TITLE
Fix dindctl script and update balenaos-in-container

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: patch
Closes: #1306
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: dindctl [hunk 1]
```diff
@@ -150,23 +172,32 @@ function runDind {
 		echo "$SUPERVISOR_IMAGE not available locally, pulling"
 		docker pull $SUPERVISOR_IMAGE
 	fi
+
+	# Save the requested image into a tar file
+	mkdir -p "${DIND_DIR}/config" && docker save --output "${DIND_DIR}/config/supervisor-image.tar" "${SUPERVISOR_IMAGE}"
+	IFS=":"
```
Comment (xginn8) [inline dindctl:178]: i would restructure this like so:

```
echo "${SUPERVISOR_IMAGE}" | awk -F: '{print "SUPERVISOR_IMAGE="$1"\nSUPERVISOR_TAG="$2"\nLED_FILE=/dev/null"}' > "./supervisor.conf"

File: dindctl [hunk 2]
```diff
@@ -178,7 +209,7 @@ function cleanDind {
 }
 
 function logs {
-	docker exec -ti balena-container-$CONTAINER_NAME journalctl $@
+	docker exec -ti balena-container-$CONTAINER_NAME journalctl -f $@
```
Comment (xginn8) [inline dindctl:212]: This'll of course follow by default, which is fine, just different from https://github.com/balena-io/balena-supervisor/pull/1307/files#diff-7e4e25b0630cffe29fc584f7a52c17bcR19

File: dindctl [hunk 3]
```diff
@@ -150,23 +172,32 @@ function runDind {
...
+		--image "${DIND_IMAGE}" \
+		--id "${CONTAINER_NAME}" \
+		--extra-args "${SUPERVISOR_DIND_MOUNTS}"
+
 }
 
 function stopDind {
 	echo "Stopping dind supervisor"
-	make -C "$SUPERVISOR_BASE_DIR" CONTAINER_NAME="$CONTAINER_NAME" stop-supervisor
+	docker stop balena-container-${CONTAINER_NAME} > /dev/null || true
```
Comment (xginn8) [inline dindctl:199]:
```suggestion
	docker stop balena-container-${CONTAINER_NAME} > /dev/null 2>&1 || true
```

File: dindctl [hunk 4]
```diff
@@ -150,23 +172,32 @@ function runDind {
...
+	docker rm -f --volumes balena-container-${CONTAINER_NAME} > /dev/null || true
```
Comment (xginn8) [inline dindctl:200]:
```suggestion
	docker rm -f --volumes balena-container-${CONTAINER_NAME} > /dev/null  2>&1 || true
```

**Comment by CameronDiver** [comment]
@balena-ci retest

**Pull request merged**